hi-ml-histopathology/src/histopathology/configs/classification/DeepSMILEPanda.py

```diff
@@ -76,6 +77,8 @@ def __init__(self, **kwargs: Any) -> None:
     def setup(self) -> None:
         if self.encoder_type == SSLEncoder.__name__:
             self.downloader = self.download_ssl_checkpoint(innereye_ssl_checkpoint_binary)
+        if self.is_finetune:
+            self.batch_size = 2
```

Comment:
why fixing the batch size here? I see two issues:
- this is an unexpected counter effect of setup - why setup should overwrite one of the input parameter?
- the optimal batch_size will depend on a number of factors (eg number of tiles, tiles size, instance memory), we shouldn't hardcode one.

An alternative solution can be in the init to suggest a certain batch_size - given number of tiles, tiles size, instance memory

Reply:
"this didn't override the original batch_size when is_finetune was set" for my own understanding can you explain why? I guess this is a problem that doesn't affect only batch_size, does it mean we need to change something in the constructor to make sure params are correctly passed?

Reply:
I wanted that whenever --is_finetune is set, it automatically takes batch_size=2 and overrides other batch sizes, else it will give OOM. I tried using this condition in the constructor, but it didn't override the batch size if we give a different batch size in the command line.

Reply:
@kenza-bouzid I think I may have tried both as I was trying to work this out, but I've noted your suggestion :)